nixos/modules/services/hardware/upower.nix

```diff
@@ -54,6 +220,24 @@ in
 
     systemd.packages = [ cfg.package ];
 
+    systemd.services.upower.environment.UPOWER_CONF_FILE_NAME = pkgs.writeTextFile {
```

Comment:
We have to be careful about this because /etc/UPower/UPower.conf could be used in some config settings (via ```environment.etc."UPower/UPower.conf"```.)

Reply:
The alternative would be to create the file at this location, which would be more convenient if people wanted to inspect this result.

However either way this should have a release note to notify people of the new module and to use it to manager upower's configuration.

Reply:
Should I change it to this?

```
systemd.services.upower.environment.UPOWER_CONF_FILE_NAME = "/etc/UPower/UPower.conf";
environment.etc."UPower/UPower.conf".text = generators.toINI {} {
  UPower = {
    EnableWattsUpPro = cfg.enableWattsUpPro;
    NoPollBatteries = cfg.noPollBatteries;
    IgnoreLid = cfg.ignoreLid;
    UsePercentageForPolicy = cfg.usePercentageForPolicy;
    PercentageLow = cfg.percentageLow;
    PercentageCritical = cfg.percentageCritical;
    PercentageAction = cfg.percentageAction;
    TimeLow = cfg.timeLow;
    TimeCritical = cfg.timeCritical;
    TimeAction = cfg.timeAction;
    CriticalPowerAction = cfg.criticalPowerAction;
  };
};
```

Reply:
I have a stronger preference for having the file only at `/etc`.
It seems `UPOWER_CONF_FILE_NAME` was implemented for their integration testing.

Also, did you need to set the file name to that path manually?
It should be compiled it to read the configuration from this directory.

Reply:
On NixOS, UPower loads its config from its store path by default. (`/nix/store/wr1699ff89pp60fcfbv42ixw8ci8h9m9-upower-0.99.11/etc/UPower/UPower.conf` on my computer). That's why I needed to set `UPOWER_CONF_FILE_NAME`.